src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -674,7 +674,8 @@ private SeparatedSyntaxList<SwitchExpressionArmSyntax> ParseSwitchExpressionArms
         {
             // PROTOTYPE(patterns2): Error recovery here leaves much to be desired.
             var arms = _pool.AllocateSeparated<SwitchExpressionArmSyntax>();
-            do
+
+            while (this.CurrentToken.Kind != SyntaxKind.CloseBraceToken)
```

Comment:
:memo: Tagging @gafter for review (small compiler change in this IDE PR).
Previously, we'd parse `e switch { }` with one switch express arm (with all missing parts). With this change, we just produce an empty list of arms.